### DISCUSSION TITLE
JSInterop error

### DISCUSSION THREAD

**Discussion by KnightCrawler72** [elsa-workflows/elsa-studio#43] [Q&A]
@sfmskywalker 
I've been working with Elsa for a couple months now, and I have to say I'm impressed with what you've built!  I'm currently trying to get the blazor UI running against my Elsa 3.0 server.  I've run npm install and build in ClientLib, and I'm able to run both the WASM and Server projects and get a list of workflow definitions.  However, when I attempt to load a definition I get the following error.  Any idea what steps I might be missing?

`JSException: Failed to fetch dynamically imported module: https://localhost:7113/_content/Elsa.Studio.DomInterop/dom.entry.js TypeError: Failed to fetch dynamically imported module: https://localhost:7113/_content/Elsa.Studio.DomInterop/dom.entry.js`

**Comment by sfmskywalker** [comment]
## UPDATE

In the below steps, I forgot to mention that in addition to running `npm install`, one must also run `npm run build`. Here are the updated instructions:

1. **Run npm install AND npm run build:**
   Navigate to the `ClientLib` directory in **both** the following projects and run `npm install` and then `npm run build`:
   - `Elsa.Studio.DomInterop`
   - `Elsa.Studio.Workflows.Designer`

2. **Build the Projects:**
   After running `npm install` AND `nom run build`, ensure to build both projects to include all necessary files in the build output.

---

Hello @amilligan72,

Thank you for your kind words!

Please ensure to follow these steps sequentially:

1. **Run npm install:**
   Navigate to the `ClientLib` directory in **both** the following projects and run `npm install`:
   - `Elsa.Studio.DomInterop`
   - `Elsa.Studio.Workflows.Designer`

2. **Build the Projects:**
   After running `npm install`, ensure to build both projects to include all necessary files in the build output.

3. **Rebuild the Solution:**
   If the issue persists, try rebuilding the entire solution. This step ensures that all the contents and dependencies are properly aligned and included in the build output.

If, after following these steps, you're still encountering issues, please let me know and we'll figure it out.

**Reply by abhishekp21** [reply]
@sfmskywalker Even after following 3 steps which you have mentioned still facing the same issue.

**Reply by sfmskywalker** [reply]
Can you make a recording and show me? There must be a step that I missed to tell you, but I can’t think of which one it may be. I looked at the build script and performed the steps manually myself to ensure my instructions are complete, but clearly they aren’t.

**Reply by abhishekp21** [reply]
https://github.com/elsa-workflows/elsa-studio/assets/18423249/03b92fa7-bfbd-48cb-ad2a-984ed149da73

@sfmskywalker let me know if any steps I'm missing

**Reply by sfmskywalker** [reply]
Hi @abhishekp21 , thank you for creating the recording, everything is clear now. In addition to `npm install`, you also need to run `npm run build`. My apologies for not having caught this sooner. I will update my instructions above, where I omitted this step.

**Reply by abhishekp21** [reply]
Thanks, It worked after running the npm run build.
